### PR TITLE
Feat: v2 embedding

### PR DESCRIPTION
**Pull request by diegog-sf**
Why: It seems that the V2 embedding was not applied to the plugin file, Just to the manifest. 

What: This PR adds support for v2 embedding.

Documentation:

https://flutter.dev/docs/development/packages-and-plugins/plugin-api-migration
https://csdcorp.com/blog/coding/handling-permission-requests-in-android-for-flutter-plugins/

Related issues:

#326

### CODE REVIEW

**Review by thomassth — APPROVED** [review]

**Comment by diegog-sf** [comment]
@thomassth Thank you for the review, is there something else needed to get this merged?

**Comment by diegog-sf** [comment]
Hello, @thomassth can you help me getting this merged?

**Comment by thomassth** [comment]
Was planning to try it once on my app before merging, in case there are some problem migrating from older versions.

But haven't got the time yet. If you have done similar things I think I can just go ahead.

**Comment by diegogarciar** [comment]
Ok I’ll do more testing on different devices and let you know. 
To the ones looking at this conversation, please try testing the PR too and let us know your findings

**Comment by thomassth** [comment]
Fixed in #336. Although still waiting for other maintainers to review and merge.

We are also looking for new maintainers! Leave a note in #339 if you’re interested

**Comment by Zookey** [comment]
I have side-loaded the dev branch with this changes and I have built release Android APK and it can not add events to the calendar anymore. It works fine when I run debug build of the Android app.

**Comment by thomassth** [comment]
> I have side-loaded the dev branch with this changes and I have built release Android APK and it can not add events to the calendar anymore. It works fine when I run debug build of the Android app.

I have just tried it on my own project and it added events just fine.

Also, please open a separate issue as this PR is not related to your topic.

**Comment by Zookey** [comment]
Thanks, I will test it once more and create a new issue. Have you tested with release APK build, not debug?

**Pull request merged**
